### PR TITLE
Fix LayerNorm.

### PR DESCRIPTION
**Pull request by tony2037**
According to "attention is all you need", the formula
apply to sublayer is supposed to be LayerNorm(x + Sublayer(x)).

As mentioned in
[issue#142](https://github.com/jadore801120/attention-is-all-you-need-pytorch/issues/142),
this implementation results from the consideration of the problem in pytorch(see [pytorch
issue#4320](https://github.com/pytorch/pytorch/issues/4320)), which has
been fix in [Fix standard deviation gradient](https://github.com/pytorch/pytorch/pull/9238).

related experience statistics
```
python train.py -data_pkl m30k_deen_shr.pkl -log m30k_deen_shr -embs_share_weight -proj_share_weight -label_smoothing -save_model trained -b 128 -warmup 12800 -epoch 40

Epoch 39
* original version
- (Training)   ppl:  355.27087, accuracy: 13.232 %, elapse: 0.691 min
- (Validation) ppl:  284.94428, accuracy: 11.988 %, elapse: 0.007 min

* follow formula in the paper
- (Training)   ppl:  297.18417, accuracy: 15.274 %, elapse: 0.696 min
- (Validation) ppl:  1478.94354, accuracy: 11.988 %, elapse: 0.007 min
```
co-contributor: @yichung279

### CODE REVIEW

**Comment by jadore801120** [comment]
Hi @zyex030640417 , thanks for the pr.
so what is the correct pytorch version you are using? 
thanks.

**Comment by tony2037** [comment]
The whole experience is based on `pytorch==1.5.0`.
BR

**Comment by VP-0822** [comment]
Hi @jadore801120 & @tony2037 , So earlier version (< 1.5) didn't experience this issue with NaN gradients? Does it only happen in the case of torch version > 1.5?  I am currently using attention mechanisms in same way and experiencing the NaN gradient issue.
